CUJExecutor/dataAggregation/Utility.java

```diff
@@ -0,0 +1,236 @@
+package dataAggregation;
+import java.io.File;  // Import the File class
+import java.io.FileNotFoundException;  // Import this class to handle errors
+import java.util.Scanner; // Import the Scanner class to read text files
+import java.util.*;
+import java.io.*;
+
+
+class Utility {
+
+	/**
+	 * Provides the info necessary to identify a section of a video (a clip)
+	 */
+   static class clipInfo {
+	String folder; //the folder where the clip is located (should be called "full_video.mp4)
+	String start; //the time at which this clip begins in the video (in the format HH:MM:SS.sss)
+	String end; //the time at which this clip ends in the video (in the format HH:MM:SS.sss)
+
+	public clipInfo (String fldr, String timestamps) {
+		folder = fldr;
+		String[] startEnd = timestamps.split("\\s*,\\s*");
+		start = startEnd[0];
+		end = startEnd[1];
+	}
+	public String getFolder() {
+		return folder;
+	}
+	public String getStart() {
+		return start;
+	}
+
+	public String getEnd() {
+		return end;
+	}
+   }
+   
+
+
+   public static Integer sum(List<Integer> list) {
+	   int sum = 0;
+	   for (int n : list) {
+		   sum += n;
+	   }
+	   return sum;
+   }
+
+   public static Integer average(List<Integer> list) {
+	   if (list.size() == 0) return null;
+	   return (sum(list) / list.size());
+   }
+
+ /**
+   * calculates the average of each column
+   * ith entry of returned array is average of ith column of input
+   */
+    public static List<Integer> averageColumns(List<List<Integer>> mat) {
+	if (mat.size() == 0) return null;
+	List<Integer> averages = new ArrayList<>();
+        for (int j = 0; j < mat.get(0).size(); j++) {
+            int sum = 0;
+            for (int i = 0; i < mat.size(); i++) {
+                sum += mat.get(i).get(j);
+            }
+            averages.add(sum / mat.size());
+        }
+        return averages;
+    }
+
+    /**
+     * Returns the median of the provided list
+     * If list is even, returns the greater of the two middle elements
+     * if list is empty, return null
+     */
+    public static Integer median(List<Integer> list) {
+	    if (list.size() == 0) return null;
+	    Collections.sort(list);
+	    return list.get(list.size() / 2);
```

Comment:
This would have the side effect of sorting the passed-in list.  May want to make a copy before sorting.